File: common/src/main/scala/com/goexp/common/util/TimerUtil.java
package com.goexp.common.util;

import static java.lang.System.out;

public class TimerUtil {

    private static TimerUtil defaults = new TimerUtil();
    private long old;

    public static TimerUtil getInstance() {
        return defaults;
    }

    private TimerUtil() {

    }

    public void start() {
        old = System.currentTimeMillis();
    }

    public void stop() {
        var nextNow = System.currentTimeMillis();
//        pool.shutdownNow();

        out.format("\nUse %d ms\n", nextNow - old);
    }
}

File: common/src/main/scala/com/goexp/common/util/TimerUtil.scala
package com.goexp.common.util


object TimerUtil {

  def apply(): TimerUtil = new TimerUtil()
}

class TimerUtil {

  private var old = 0L

  def start() = {
    old = System.currentTimeMillis
    this
  }


  def stop(): Unit = {
    val nextNow = System.currentTimeMillis
    print(s"\nUse ${nextNow - old} ms\n")
  }

}